src/wallet/wallet2.cpp

```diff
@@ -575,6 +575,27 @@ void wallet2::check_acc_out_precomp(const crypto::public_key &spend_public_key,
   error = false;
 }
 //----------------------------------------------------------------------------------------------------
+void wallet2::check_acc_out_precomp_onetime(const crypto::public_key &spend_public_key, const tx_out &o, const std::pair<crypto::secret_key, crypto::public_key>& derivation_info, const crypto::secret_key* onetime_c, size_t i, is_received_info &received, uint64_t &money_transfered, bool &error) const
+{
+  const crypto::secret_key &view_secret_key = derivation_info.first;
+  const crypto::public_key &tx_public_key   = derivation_info.second;
+  crypto::key_derivation derivation;
+  crypto::generate_key_derivation(tx_public_key, view_secret_key, derivation);
+  check_acc_out_precomp(spend_public_key, o, derivation, i, received.standard, money_transfered, error);
+  if (!received.standard && onetime_c)
+  {
+    crypto::secret_key d;
+    crypto::public_key D;
+    crypto::hash_to_scalar(onetime_c->data, HASH_SIZE, reinterpret_cast<crypto::ec_scalar&>(d));
+    secret_key_to_public_key(d, D);
+    crypto::public_key spend_public_key2;
+    crypto::key_derivation derivation2;
+    crypto::add_public_key(spend_public_key, D, spend_public_key2);
+    crypto::generate_key_derivation(tx_public_key, *onetime_c, derivation2);
+    check_acc_out_precomp(spend_public_key2, o, derivation2, i, received.onetime, money_transfered, error);
+  }
```

Comment:
No, it is not bad practice per se. I can't see where that code is because github won't let me  see more context, but if it's not a local variable (and thus you know it was initialized), then it should be set, because the caller may have done whatever with it before. If it's a local var declared just above, then it's fine as is. But in general, I like inits to be explicit when a test is done. Saves on uncertainty and bugs if/when code changes elsewhere.